File: sub-mysql-web/src/main/java/org/github/base/entity/SysRoleMenuEntity.java
package org.github.base.entity;

import org.github.base.Model;
import com.baomidou.mybatisplus.annotation.*;
import java.io.Serializable;
import javax.annotation.Nullable;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * <p>
 * 角色与菜单对应关系
 * </p>
 *
 * @author JYD_XL
 * @since 2021-05-20
 */
@SuppressWarnings("all")
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_role_menu")
@ApiModel(value="SysRoleMenuEntity对象", description="角色与菜单对应关系")
public class SysRoleMenuEntity extends Model<SysRoleMenuEntity> {

  /** UID */
  private static final long serialVersionUID = 1L;

  @TableId(value = "id", type = IdType.AUTO)
  private @Nullable Long id;

  /** 角色ID */
  @ApiModelProperty(value = "角色ID")
  @TableField("role_id")
  private @Nullable Long roleId;

  /** 菜单ID */
  @ApiModelProperty(value = "菜单ID")
  @TableField("menu_id")
  private @Nullable Long menuId;

  @Override
  public Serializable pkVal() {
    return this.id;
  }

}

File: sub-mysql-mydb-entity/src/main/java/org/github/mysql/mydb/base/entity/SysRoleMenuEntity.kt
@file:Suppress("UnusedImport", "unused")

package org.github.mysql.mydb.base.entity

import com.baomidou.mybatisplus.annotation.IdType
import com.baomidou.mybatisplus.annotation.TableField
import com.baomidou.mybatisplus.annotation.TableId
import com.baomidou.mybatisplus.annotation.TableName
import java.io.Serializable
import org.github.base.Entity

/**
 * 角色-菜单关系表
 *
 * @author JYD_XL
 */
@TableName("sys_role_menu")
open class SysRoleMenuEntity : Entity() {

    /** 主键 */
    @TableId(value = "id", type = IdType.ASSIGN_ID)
    open var id: Long? = null

    /** 角色id */
    @TableField("role_id")
    open var roleId: Long? = null

    /** 菜单id */
    @TableField("menu_id")
    open var menuId: Long? = null

    companion object {

        private const val serialVersionUID = 1L

        const val ID : String = "id"

        const val ROLE_ID : String = "role_id"

        const val MENU_ID : String = "menu_id"

    }

    override fun pkVal(): Serializable? {
        return id
    }

    override fun toString(): String {
        return "SysRoleMenuEntity{" +
        "id=" + id +
        ", roleId=" + roleId +
        ", menuId=" + menuId +
        "}"
    }

}
